x-pack/plugins/security_solution/server/lists_integration/endpoint/validators/blocklist_validator.ts

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ENDPOINT_BLOCKLISTS_LIST_ID } from '@kbn/securitysolution-list-constants';
+import { schema, TypeOf } from '@kbn/config-schema';
+import { ExceptionListItemSchema } from '@kbn/securitysolution-io-ts-list-types';
+import { OperatingSystem } from '@kbn/securitysolution-utils';
+import { BaseValidator } from './base_validator';
+import { ExceptionItemLikeOptions } from '../types';
+import {
+  CreateExceptionListItemOptions,
+  UpdateExceptionListItemOptions,
+} from '../../../../../lists/server';
+import { isValidHash } from '../../../../common/endpoint/service/trusted_apps/validations';
+import { EndpointArtifactExceptionValidationError } from './errors';
+
+const FileHashField = schema.oneOf([
+  schema.literal('file.hash.md5'),
+  schema.literal('file.hash.sha1'),
+  schema.literal('file.hash.sha256'),
+]);
+const FileExecutablePath = schema.literal('file.executable.caseless');
+const FileCodeSigner = schema.literal('file.Ext.code_signature');
+
+const ConditionEntryTypeSchema = schema.literal('match_any');
+const ConditionEntryOperatorSchema = schema.literal('included');
+
+type ConditionEntryFieldAllowedType =
+  | TypeOf<typeof FileHashField>
+  | TypeOf<typeof FileExecutablePath>
+  | TypeOf<typeof FileCodeSigner>;
+
+type BlocklistConditionEntry =
+  | {
+      field: ConditionEntryFieldAllowedType;
+      type: 'match_any';
+      operator: 'included';
+      value: string[];
+    }
+  | TypeOf<typeof WindowsSignerEntrySchema>;
+
+/*
+ * A generic Entry schema to be used for a specific entry schema depending on the OS
+ */
+const CommonEntrySchema = {
+  field: schema.oneOf([FileHashField, FileExecutablePath]),
+  type: ConditionEntryTypeSchema,
+  operator: ConditionEntryOperatorSchema,
+  // If field === HASH then validate hash with custom method, else validate string with minLength = 1
+  value: schema.conditional(
+    schema.siblingRef('field'),
+    FileHashField,
+    schema.arrayOf(
```

Comment:
~These array's of values needs to have at least a length of 1, right? and for hashes, no more than 3? so should the min/max options to `.arrayOf()` be used here?~

never mind - I see it further below